src/util/expr.h

```diff
@@ -59,9 +59,18 @@ class exprt:public irept
   // constructors
   exprt() { }
   explicit exprt(const irep_idt &_id):irept(_id) { }
-  exprt(const irep_idt &_id, const typet &_type):irept(_id)
+
+  exprt(irep_idt _id, typet _type)
+    : irept(std::move(_id), {{ID_type, std::move(_type)}}, {})
+  {
+  }
+
+  exprt(irep_idt _id, typet _type, operandst &&_operands)
+    : irept(
+        std::move(_id),
+        {{ID_type, std::move(_type)}},
```

Comment:
moving a `dstring` may well be slower than copying it.

Reply:
Will remove.

Reply:
Or actually: shouldn't we just define a move constructor for `dstringt` that actually is cheap? For the case that `irep_idt == std::string` we'd still want the `std::move`.

Reply:
I have added a commit that adds efficient move construction/assignment to `dstringt`.

Reply:
But we don't know whether `irep_idt` is a `std::string` (where it would seem worthwhile) or a `dstringt` (where it's certainly unnecessary). But also that isn't something I want this PR to get blocked on, I'm happy to follow whatever I'm advised to do.